server/src/features/facilities/facility.service.js

```diff
@@ -1,70 +1,123 @@
 import { CourtBooking, GymSession } from "./facility.model.js";
-import { sendGymSessionCancellationEmail, sendGymSessionUpdateEmail } from "../auth/email.service.js";
+import {
+  sendGymSessionCancellationEmail,
+  sendGymSessionUpdateEmail,
+} from "../auth/email.service.js";
 
 class FacilitiesServiceClass {
-async getCourtSchedules() {
-  const today = new Date();
-  today.setHours(0, 0, 0, 0);
+  async reserveCourt(userId, reservationData) {
+    const { courtType, date, startTime, endTime } = reservationData;
 
-  const weekAhead = new Date(today);
-  weekAhead.setDate(today.getDate() + 6);
+    // Validate availability
+    const schedules = await this.getCourtSchedules();
+    const courtSchedule = schedules[courtType];
+
+    if (!courtSchedule) {
+      throw new Error(`Invalid court type: ${courtType}`);
+    }
+
+    const bookingDate = new Date(date);
```

Comment:
```suggestion
    const bookingDate = new Date(date);
    const today = new Date();
     today.setHours(0,0,0,0);

     if (bookingDate < today) {
        throw new Error("Cannot book a past date");
}

```
Validate date must be in the future